api/client/node/test/next.test.ts

```diff
@@ -0,0 +1,75 @@
+import { describe, it, expect, vi } from 'vitest'
+import { createOpenAIStreamCallback } from '../dist/index.js'
+import type { OpenAIStreamCallbacks } from 'ai'
+
+describe('next', () => {
+  const model = 'gpt-4'
+  const prompts = ['Say Hello World!']
+
+  describe('createOpenAIStreamCallback', () => {
+    it('should return usage', () =>
+      new Promise<void>((done) => {
+        const streamCallbacks = createOpenAIStreamCallback(
+          {
+            model,
+            prompts,
+          },
+          {
+            onUsage: (usage) => {
+              expect(usage).toEqual({
```

Comment:
You might need to wrap this into a try/catch and reject the promise?

Reply:
It exists this way too